### PR TITLE
Update readme

### PR DESCRIPTION
**Pull request by ShiriNmi1520**
## Changes

* Update basic usage example code

* Fixed some typos

* The other example code such as ```registerTagEvent```, ```NDEF-Writing``` and ```Mifare Ultralight``` might be improved in future

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -80,59 +80,78 @@ We have published a React Native NFC course with [newline.co](https://www.newlin
...
-import React from 'react';
-import {View, Text, TouchableOpacity, StyleSheet} from 'react-native';
-import NfcManager, {NfcTech} from 'react-native-nfc-manager';
-
-// Pre-step, call this before any NFC operations
-NfcManager.start();
-
-function App() {
-  async function readNdef() {
+import { StatusBar } from 'expo-status-bar';
```
Comment (whitedogg13) [inline README.md:83]: @ShiriNmi1520 it's awesome! I think we can remove this `expo-status-bar` stuff in case our user uses `react-native-cli`

Comment (ShiriNmi1520) [reply README.md:83]: Sure I'll commit a change, hang on

**Comment by whitedogg13** [comment]
It seems that the code snippet want to demonstrate how to start NFC scanning automatically without pressing a button, but I think it will be easier to understand if the whole process starts from a button press. 

In such a case, we can avoid extra `useEffect` hook and focus our code on `NFC` part, since it serves as the most basic example for the library.

About the `scan without button press` scenario, I think it might deserve another README section since people often ask about it.

**Comment by ShiriNmi1520** [comment]
So it might be better to move this part to the new MD file and give the link in the original README right?

**Comment by whitedogg13** [comment]
Yes, I think this will be great!

**Comment by ShiriNmi1520** [comment]
Ok I'll mark this PR as a draft till I have done with this part

**Comment by whitedogg13** [comment]
Both new MD or new section are good to me

**Comment by ShiriNmi1520** [comment]
I'd prefer new section

**Comment by ShiriNmi1520** [comment]
> It seems that the code snippet wants to demonstrate how to start NFC scanning automatically without pressing a button, but I think it will be easier to understand if the whole process starts from a button press.
> 
> In such a case, we can avoid the extra `useEffect` hook and focus our code on the `NFC` part, since it serves as the most basic example for the library.
> 
> About the `scan without button press` scenario, I think it might deserve another README section since people often ask about it.

About extra ```useEffect```, did you mean the following code about initializing nfcManager?
I implement ```nfcManager.start()``` this way to prevent multi time initalize, or there's might be better way to implement this
```javascript
  // Initialize the NFC manager (This will run only once)
  useEffect(() => {
    nfcManager.isSupported().then(supported => {
      if (supported) {
        nfcManager.start();
      } else {
        setNfcData('NFC not supported');
      }
    })
  }, [])
```

**Comment by whitedogg13** [comment]
I understand you're using a standard `zero deps` pattern to initialize the native module. 

The problem is inside the next hook:
```javascript
  // Listen to NFC events (This will run every-time after nfcData changes)
  useEffect(() => {
    getNFCInfo();
  }, [nfcData])
```

Two points here:
- Once `nfcData` changed, the `getNFCInfo()` will be called, which is generally not desired.
- It cannot garuantee native `NfcManager.start()` is successfully called

And one extra point to be aware is that: `NfcManager.start()` is also `async` call, so there might be a chance that you trigger the NFC scan before the native module is fully initialized.

**Comment by ShiriNmi1520** [comment]
> I understand you're using a standard `zero deps` pattern to initialize the native module.
> 
> The problem is inside the next hook:
> 
> ```js
>   // Listen to NFC events (This will run every-time after nfcData changes)
>   useEffect(() => {
>     getNFCInfo();
>   }, [nfcData])
> ```
> 
> Two points here:
> 
> * Once `nfcData` changed, the `getNFCInfo()` will be called, which is generally not desired.
> * It cannot garuantee native `NfcManager.start()` is successfully called
> 
> And one extra point to be aware is that: `NfcManager.start()` is also `async` call, so there might be a chance that you trigger the NFC scan before the native module is fully initialized.

I got it, thanks for your feedback, appreciated :)

**Pull request closed without merging**